src/apps/pillarx-app/components/TrendingTokens/TrendingTokens.tsx

```diff
@@ -0,0 +1,92 @@
+import { useTranslation } from 'react-i18next'
+import { createRef, useEffect, useState } from 'react';
+
+// types
+import { TrendingTokenData } from '../../../../types/api'
+
+// hooks
+import useRefDimensions from '../../hooks/useRefDimensions';
+
+// images
+
+// components
+import TileContainer from '../TileContainer/TileContainer'
+import Body from '../Typography/Body';
+import TrendingTokenInfo from '../TrendingTokenInfo/TrendingTokenInfo';
+import SkeletonLoader from '../../../../components/SkeletonLoader';
+
+
+type TrendingTokensProps = {
+    data: TrendingTokenData[] | undefined;
+    isDataLoading: boolean;
+}
+
+const TrendingTokens = ({ data, isDataLoading }: TrendingTokensProps) => {
+    const [t] = useTranslation();
+
+    const [trendingTokenWidth, setTrendingTokenWidth] = useState<number>(0);
+
+    useEffect(() => {
+        const handleTrendingTokenWidthResize = () => {
+            if (window.innerWidth >= 1024) {
+                setTrendingTokenWidth(159);
+            } else if (window.innerWidth >= 800) {
+                setTrendingTokenWidth(152);
+            } else {
+                // 102px for width of the TrendingTokenInfo component 100px + 2px of space
+                setTrendingTokenWidth(102);
+            }
+        };
+
+        handleTrendingTokenWidthResize();
+        window.addEventListener('resize', handleTrendingTokenWidthResize);
+
+        return () => {
+            window.removeEventListener('resize', handleTrendingTokenWidthResize);
+        };
+    }, []);
+    
+    const divRef = createRef()
+    const dimensions = useRefDimensions(divRef as React.RefObject<HTMLDivElement>)
+
+    const numberTrendingTokens = Math.floor(dimensions.width / trendingTokenWidth);
+
+    const DUMMY_TOKEN_VALUE = 160.12345678;
+
+    const DUMMY_TOKEN_PERCENTAGE = (tokenPriceChange: number) => 
+        (tokenPriceChange / DUMMY_TOKEN_VALUE) * 100;
```

Comment:
Can i just confirm what these variables are being used for?

Reply:
The DUMMY_TOKEN_VALUE is being used for the token value in $ (until we add this in the api) and the DUMMY_TOKEN_PERCENTAGE is being calculated from the token value changes in the last 24h (which we have on the API, but then we are missing the token value) - This are placeholders essentially and will be replaced once the api is giving us the Token Value